tests/rptest/transactions/stream_verifier_test.py

```diff
@@ -0,0 +1,166 @@
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+import json
+
+from rptest.clients.types import TopicSpec
+from rptest.services.admin import Admin
+from rptest.services.cluster import cluster
+from rptest.transactions.verifiers.stream_verifier import StreamVerifier
+from rptest.tests.redpanda_test import RedpandaTest
+
+
+class StreamVerifierTest(RedpandaTest):
+    # Max time to wait for the cluster to be healthy once more.
+    HEALTHY_WAIT_SECONDS = 20 * 60
+
+    # Up to 5 min to stop the node with a lot of topics
+    STOP_TIMEOUT = 60 * 5
+
+    def __init__(self, test_context):
+        # basic topic info
+        self.source_topic = "stream_topic_src"
+        self.target_topic = "stream_topic_dst"
+
+        # Calculated speed of producer is ~100k messages per minute, which is ~1500/sec
+        # Amtomic thread processed ~250 messages per second.
+        # Idea is to have atomic processing going while producer is still producing messages
+        # This will slow down single node Stream producer, but will check several incoming
+        # requests on RP for the same topic.
+        self.default_message_count = 25000
+
+        self.num_partitions = 1
+        self.num_replicas = 3
+
+        # Consumer sleep time when EOF
+        self.consume_sleep_time_s = 10
+
+        # Save context for future use
+        self.test_context = test_context
+
+        extra_rp_conf = {
+            "id_allocator_replication": 3,
+            "default_topic_replications": 3,
+            "default_topic_partitions": 1
+        }
+        # 3 brokers active, 1 standby
+        super(StreamVerifierTest, self).__init__(num_brokers=3,
+                                                 test_context=test_context,
+                                                 extra_rp_conf=extra_rp_conf)
+
+        self.admin = Admin(self.redpanda)
+        self.installer = self.redpanda._installer
+
+    def create_topics(self):
+        # Source topics
+        self.client().create_topic(
+            TopicSpec(name=self.source_topic,
+                      partition_count=self.num_partitions,
+                      replication_factor=self.num_replicas))
+
+        self.client().create_topic(
+            TopicSpec(name=self.target_topic,
+                      partition_count=self.num_partitions,
+                      replication_factor=self.num_replicas))
+
+    def start_producer(self,
+                       wait_msg_count=50,
+                       messages_per_sec=0,
+                       timeout=30):
+        self.verifier.remote_start_produce(self.source_topic,
+                                           self.default_message_count,
+                                           messages_per_sec=messages_per_sec)
+        self.logger.info(f"Waiting for {wait_msg_count} produces messages")
```

Comment:
Is this missing a call to ensure progress or am I misreading?

Reply:
It is in the next line: https://github.com/redpanda-data/redpanda/pull/19976/files/cf47568343cf5570ff48c14aeeae5da61a87b1db#diff-54b41e2f630fa459c2026ad573cab55e04668e9ba1bf865709c134155522fc46R79